morphman/automated_landmarking/automated_landmarking_tools.py

```diff
@@ -0,0 +1,245 @@
+        print("-- Sanity check failed, checking for maximums")
+
+        value_index = coronal_coordinate[argrelextrema(coronal_coordinate, np.greater_equal)[0]].max()
+        max_coronal_coordinate_id = np.array(coronal_coordinate.tolist().index(value_index))
+        if abs(length[max_coronal_coordinate_id] - length[-1]) > search_tolerance:
+            print("-- Sanity check failed, no anterior bend in model. Exiting.")
+            sys.exit(1)
+    return max_coronal_coordinate_id
+
+
+def get_centerline_coordinates(line, length):
+    x = np.zeros(length.shape[0])
+    y = np.zeros(length.shape[0])
+    z = np.zeros(length.shape[0])
+    for i in range(z.shape[0]):
+        x[i] = line.GetPoints().GetPoint(i)[0]
+        y[i] = line.GetPoints().GetPoint(i)[1]
+        z[i] = line.GetPoints().GetPoint(i)[2]
+
+    return {"x": x, "y": y, "z": z}
+
+
+def orient_centerline(ica_centerline):
+    """
+    Check the orientation of the axial coordinate and
+    reverse centerline if path of centerline is in distal direction.
+
+    Args:
+        ica_centerline (vtkPolyData): Centerline to check
+
+    Returns:
+        ica_centerline (vtkPolyData): Centerline, possibly reversed
+    """
+    length = get_curvilinear_coordinate(ica_centerline)
+    coordinates = get_centerline_coordinates(ica_centerline, length)
+
+    # Iterate through coordinates and find axial coordinate
+    for _, coordinate in coordinates.items():
+        coordinate = gaussian_filter(coordinate, 25)
+        min_c = min(coordinate)
+        max_c = max(coordinate)
+        id0 = coordinate.tolist().index(min_c)
+        id1 = coordinate.tolist().index(max_c)
+        if id0 == 0 and id1 == (len(coordinate) - 1):
+            return ica_centerline
+
+        if id1 == 0 and id0 == (len(coordinate) - 1):
+            return reverse_centerline(ica_centerline)
+
+    return ica_centerline
+
+
+def spline_centerline_and_compute_geometric_features(line, smooth, nknots):
+    """
+    Compute attributes and geometric parameters of input
+    centerline, using B-splines (SciPy).
+
+    Args:
+        line (vtkPolyData): Centerline data.
+        smooth (bool): Smooth centerline with VMTK if True.
+        nknots (int): Number of knots for B-splines.
+
+    Returns:
+        line (vtkPolyData): Splined centerline.
+    Returns:
+        max_point_ids (ndarray): Array of max curvature values
+    Returns:
+        min_point_ids (ndarray): Array of min curvature values
+    """
+    data = np.zeros((line.GetNumberOfPoints(), 3))
+    curv_coor = get_curvilinear_coordinate(line)
+
+    # Collect data from centerline
+    for i in range(data.shape[0]):
+        data[i, :] = line.GetPoints().GetPoint(i)
+
+    t = np.linspace(curv_coor[0], curv_coor[-1], nknots + 2)[1:-1]
+
+    fx = splrep(curv_coor, data[:, 0], k=4, t=t)
+    fy = splrep(curv_coor, data[:, 1], k=4, t=t)
+    fz = splrep(curv_coor, data[:, 2], k=4, t=t)
+
+    fx_ = splev(curv_coor, fx)
```

Comment:
Why not do:

```python
data[:, 0] = splev(curv_coor, fx)
data[:, 1] = splev(curv_coor, fy)
data[:, 2] = splev(curv_coor, fz)
```
directely.

Reply:
There was use of `fx` and so on later on, but I changed it so you just fetch them from `data` instead.